langlib/lang.stream/src/main/ballerina/stream.bal

```diff
@@ -43,6 +43,12 @@ type Type1 any|error;
 
 # Selects the members from a stream for which a function returns true.
 #
+# ```ballerina
+# stream<int> scoreStream = [45, 60, 75, 30, 90].toStream();
+# stream<int> filtered = scoreStream.filter(score => score > 50);
+# filtered.next() ⇒ {"value":60}
```

Comment:
Going with the suggestion for array functions, shouldn't this be

```suggestion
# scoreStream.filter(score => score > 50).next() ⇒ {"value":60}
```

Reply:
Can't we also just use something like `strm` as the name?

Reply:
hmm. I kind of prefer having proper names. :)

Reply:
BTW, what if we have the earlier version?

```
stream<int> filtered = scoreStream.filter(score => score > 50);
filtered.next() ⇒ {"value":60}
```

I feel like it is better, something similar to below array case,

```
object {
    public isolated function next() returns record {|int value;|}?;
} iterator = [2, 4, 6, 8].iterator();
iterator.next() ⇒ {"value":2}
```

WDYT?

Reply:
> hmm. I kind of prefer having proper names. :)

Can we have something like `scores` instead of `scoreStream`?



> BTW, what if we have the earlier version?
> 
> ```
> stream<int> filtered = scoreStream.filter(score => score > 50);
> filtered.next() ⇒ {"value":60}
> ```
> 
> I feel like it is better, something similar to below array case,
> 
> ```
> object {
>     public isolated function next() returns record {|int value;|}?;
> } iterator = [2, 4, 6, 8].iterator();
> iterator.next() ⇒ {"value":2}
> ```
> 
> WDYT?

The issue with including `stream<int> filtered = scoreStream.filter(score => score > 50);` though is that we kinda expose the type param behaviour, right? Which we haven't done with others.

Reply:
> > hmm. I kind of prefer having proper names. :)
> 
> Can we have something like `scores` instead of `scoreStream`?

Hmm, OK. 
> 
> > BTW, what if we have the earlier version?
> > ```
> > stream<int> filtered = scoreStream.filter(score => score > 50);
> > filtered.next() ⇒ {"value":60}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I feel like it is better, something similar to below array case,
> > ```
> > object {
> >     public isolated function next() returns record {|int value;|}?;
> > } iterator = [2, 4, 6, 8].iterator();
> > iterator.next() ⇒ {"value":2}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > WDYT?
> 
> The issue with including `stream<int> filtered = scoreStream.filter(score => score > 50);` though is that we kinda expose the type param behaviour, right? Which we haven't done with others.

Hmm. I feel it is fine.